### ISSUE TITLE
Ubuntu Snap: Async segmentation fault when accessing powerMonitor in app.on('ready')

### ISSUE THREAD

**Issue by dimitarnestorov** [electron/electron#13767]
* Electron Version: ^2.0.0
* Operating System (Platform and Version): Ubuntu 16.04 Snap 
* Last known working Electron version: 1.8.7

**Expected Behavior**

No segmentation fault

**Actual behavior**
```
Segmentation fault (core dumped)
```
**To Reproduce**

With yarn:
```sh
git clone https://github.com/dimitarnestorov/electron-segfault.git
cd electron-segfault
yarn
yarn build
snap install dist/electron-quick-start-test_1.0.0_amd64.snap --dangerous
snap run electron-quick-start-test
```

With npm:
```sh
git clone https://github.com/dimitarnestorov/electron-segfault.git
cd electron-segfault
npm install
npm run build
snap install dist/electron-quick-start-test_1.0.0_amd64.snap --dangerous
snap run electron-quick-start-test
```

**Screenshots**

![image](https://user-images.githubusercontent.com/8790386/43044519-1013dcc8-8db0-11e8-8255-eacff2f4ddb7.png)

**Additional Information**

* The segfault doesn't happen right after https://github.com/dimitarnestorov/electron-segfault/blob/94cccfbc6b03fccf0878ba192d56f4879e5aa9cb/main.js#L6
The async queue runs a couple of times. Adding the following code after that line confirms that:
```javascript
  function a(){
    console.log('Called')
    setTimeout(a);
  }
  setTimeout(a);
```
![image](https://user-images.githubusercontent.com/8790386/43044763-60813d4a-8db5-11e8-8db5-428d63c6e385.png)
* This issue prevents me from using Sentry on Ubunutu. More specifically: https://github.com/getsentry/sentry-electron/blob/32e0e2dcb89f413a75deaa79195aea7e1aa60c8c/src/main/backend.ts#L320
* Also prevents @kspearrin from locking his password manager when the computer is suspended: https://github.com/bitwarden/desktop/commit/fea8ab5921cc7b7f8a6a0f8d37f7ae13b64fede3#diff-09666e7a76984a3319985ca3e76c4744

**Comment by sofianguy** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron v4.2.x or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons. This issue will be closed 7 days from now if there is no response.

Thanks in advance! Your help is appreciated.

**Comment by dimitarnestorov** [comment]
Since all issues that are referencing this one are closed and I just tested with 6.0.1 and it works I'm closing.
